src/OpalCompiler-Core/OCBytecodeToASTCache.class.st

```diff
@@ -57,20 +57,20 @@ OCBytecodeToASTCache >> firstBcOffset [
 ]
 
 { #category : #initialization }
-OCBytecodeToASTCache >> generateForMethodNode: aNode [
-	| methodIR currentBcOffset |
-	methodNode := aNode.
-	methodIR := methodNode ir.
+OCBytecodeToASTCache >> generateForNode: aNode [
+	| blockIR currentBcOffset |
+	methodOrBlockNode := aNode.
+	blockIR := aNode ir.
 	bcToASTMap := Dictionary new.
-	firstBcOffset := methodNode compiledMethod initialPC.
+	firstBcOffset := blockIR compiledMethod initialPC.
 
 	"if there is a primitive, we need to record the offset"
-	(methodIR irPrimitive num ~= 0
-		or: [ methodIR compiledMethod isQuick ])
-		ifTrue: [ bcToASTMap at: firstBcOffset put: methodNode ]. 
-		
+	(blockIR irPrimitive num ~= 0
+		or: [ blockIR compiledMethod isQuick ])
+		ifTrue: [ bcToASTMap at: firstBcOffset put: methodOrBlockNode ]. 
+
 	currentBcOffset := firstBcOffset.
-	methodIR startSequence withAllSuccessors
+	blockIR startSequence withAllSuccessors
```

Comment:
I am just wondering: we can seamlessly use the same algorithm for methods and full blocks because full blocks are also methods without anything specific? We do not need any specific test to validate it?

Reply:
The branch in #sourceNodeForPC: on block node that calls it is executed 77 times while running the Opal tests suite and >60 times when running all debugger tests.

It is covered by existing tests.

Reply:
of course more test would be nice